llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp

```diff
@@ -3784,6 +3784,68 @@ bool X86AsmParser::processInstruction(MCInst &Inst, const OperandVector &Ops) {
     Inst.setOpcode(X86::INT3);
     return true;
   }
+  // `{evex} cmp <>, <>` is alias of `ccmpt {dfv=} <>, <>`
+#define FROM_TO(FROM, TO)                                                      \
+  case X86::FROM: {                                                            \
+    if (ForcedOpcodePrefix == OpcodePrefix_EVEX) {                             \
```

Comment:
Can we hoist out this check in each case, e.g.,

if (ForcedOpcodePrefix == OpcodePrefix_EVEX) {
  switch (Inst.getOpcode()) {
    FROM_TO(CMP64rr, CCMP64rr)
    ...
}

Reply:
We may also need to make the code to a lambda rather than duplicate a lot of times.

Reply:
> Can we hoist out this check in each case, e.g.,
> 
> ```
> if (ForcedOpcodePrefix == OpcodePrefix_EVEX) {
>   switch (Inst.getOpcode()) {
>     FROM_TO(CMP64rr, CCMP64rr)
>     ...
> }
> ```

Sorry, I don't quite understand the benefit. Even without hoisting, at runtime `if (ForcedOpcodePrefix == OpcodePrefix_EVEX)` will be checked once, and it can utilize the current `Inst.getOpcode()` above.

Reply:
This is to reduce code size. Because the check will be duplicated multi times.

Reply:
Maybe we don't need to care about the code size after macro expansion. Before expansion, the code is not duplicated and easy to maintain. The compiler used to build LLVM should be able to extract the code into a function if it's beneficial.